core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -141,8 +141,8 @@ public CallOptions withRequestKey(@Nullable RequestKey requestKey) {
    */
   @ExperimentalApi
   @Nullable
-  public RequestKey getRequestKey() {
-    return requestKey;
+  public Attributes getAttributes() {
```

Comment:
It seems either way we would need a `getAttributes()` method. Even if we have `withAttribute(key, value)` it seems we would still need `withAttributes(attributes)` in order to set whole-sale and replace. I'm not sure how many users will use the method, so I'm not sure if the convenience `withAttribute(key, value)` is necessary. I would agree it is a nicer though. I don't mind much either way.

`affinityData` SGTM. I would like something more concrete than "Attributes," especially since it isn't necessarily "call option attributes."